### PR TITLE
Modernized help UI prototype: Refactor JavaScript code

### PR DESCRIPTION
**Pull request by howlger**
Refactor JavaScript code for better readability. No functional changes.

See also https://github.com/eclipse-platform/eclipse.platform/pull/735#pullrequestreview-1666671985

### CODE REVIEW

File: ua/org.eclipse.help.webapp/m/index.js [hunk 1]
```diff
@@ -99,20 +105,20 @@
                          var scopeName = decodeHtml(match[1]);
                          if (scopeName.substring(0, 1) == '\u200B') {
                              if (scopeName.length < 3) {
-                                 searchScope.l = scopeName.length;
+                                 searchScope.level = scopeName.length;
                                  break;
                              } else {
-                                 searchScope.l = 4;
-                                 searchScope.t = scopeName.length - 3;
+                                 searchScope.level = SEARCH_SCOPE_LEVEL_CUSTOM;
```
Comment (iloveeclipse) [inline ua/org.eclipse.help.webapp/m/index.js:111]: SEARCH_SCOPE_LEVEL_CUSTOM is 3 now (was 4). Expected?

Comment (howlger) [reply ua/org.eclipse.help.webapp/m/index.js:111]: Yes. 3 was reserved for a future use that never happened. Now it is a constant that is used as an enum value. I tested it manually by setting these scope level constants to random numbers to be sure not to miss something.

**Comment by howlger** [comment]
The test failures do not seem to me to be related to this change.

**Comment by howlger** [comment]
To review this change, the modernized help UI has to be enabled via the Java property `-Dorg.eclipse.help.webapp.experimental.ui=true`

**Comment by iloveeclipse** [comment]
@howlger : is this still relevant?

**Comment by howlger** [comment]
> @howlger : is this still relevant?

Yes, as it blocks https://github.com/eclipse-platform/eclipse.platform/pull/735. We at Advantest are using this change for quite some time now without any problems.

**Comment by iloveeclipse** [comment]
> We at Advantest are using this change for quite some time now without any problems.

:-) 

If so, let's merge

**Pull request merged**